### ISSUE TITLE
Please migrate your plugin to Nativescript 7

### ISSUE THREAD

**Issue by cjohn001** [DeepakArora76/nativescript-dna-deviceinfo#8]

**Comment by limux82** [comment]
Is it someone working in migrate the project to NS7. 
I've tried to migrate, but always an error on  this class

@NativeClass()
class BluetoothManagerDelegate extends NSObject implements CBCentralManagerDelegate {
  // Note: This ObjCProtocols is needed.
  public static ObjCProtocols = [CBCentralManagerDelegate];

  private resolve: any;

  static new(): BluetoothManagerDelegate {
    return <BluetoothManagerDelegate>super.new();
  }

  initWithResolverCallback(resolve: any) {
    this.resolve = resolve;
    return this;
  }

**The error is
 error TS2304: Cannot find name 'NativeClass'.**

I migrated other 7 plugins, but I don't have this error

**Comment by mm-psymbl** [comment]
same problem here :( any hope?

**Comment by DeepakArora76** [comment]
Hi guys. I am running busy with other projects these days. I would like to support you all. I do not promise anything, but I will try.

**Comment by DeepakArora76** [comment]
@mm-psymbl @limux82 I have upgraded the plugin to NS 7.0. Please note that I have not tested on iOS and Android devices. The latest version is 3.0.0. The code is also updated as well. 

I appreciate if you could spare some time in testing the plugin.

Have a great morning and Sunday :)

**Comment by neil-119** [comment]
Hey, thanks a lot of making this awesome plugin and updating to NS 7. Unfortunately we are still running into this error on the latest package version 3.3.1 on iOS 14.3.

```
***** Fatal JavaScript exception - application has been terminated. *****
2021-03-04 15:45:16.278662-0600
NativeScript encountered a fatal error: Uncaught ReferenceError: NativeClass is not defined
at 
	../node_modules/nativescript-dna-deviceinfo/deviceinfo.js
```

**Comment by DeepakArora76** [comment]
@neil-119  I am glad to know that you are using the plugin. The problem which you have described is not reproducible
 at my end. 

I have a demo repository that you can try out and check if the same problem persists at your end. The link to the repo is: https://github.com/DeepakArora76/JSDeviceInfoDemo

It is really hard for me to see the problem. Supplementary information will help me diagnose the hidden issue.

- What is the framework that you are targeting? 
- What does your development env look like?

In the meanwhile, I like you to update your toolchain, and the OS and Xcode. 

I have Xcode 12.4, Build version 12D4e. The OS is MacOS Big Sur Version 11.2.1.

Below is the output of the  **ns doctor** command.

Deepak$ ns doctor
You have both a nativescript.config.js and nativescript.config.ts file. Defaulting to nativescript.config.ts.
✔ Getting environment information 

No issues were detected.
✔ Your ANDROID_HOME environment variable is set and points to correct directory.
✔ Your adb from the Android SDK is correctly installed.
✔ The Android SDK is installed.
✔ A compatible Android SDK for compilation is found.
✔ Javac is installed and is configured properly.
✔ The Java Development Kit (JDK) is installed and is configured properly.
✔ Xcode is installed and is configured properly.
✔ xcodeproj is installed and is configured properly.
✔ CocoaPods are installed.
✔ CocoaPods update is not required.
✔ CocoaPods are configured properly.
✔ Your current CocoaPods version is newer than 1.0.0.
✔ Python installed and configured correctly.
✔ The Python 'six' package is found.
✔ Xcode version 12.4.0 satisfies minimum required version 10.
✔ Getting NativeScript components versions information...


I will be looking forward to your reply.

**Comment by DeepakArora76** [comment]
@neil-119  Any update on your progress?

**Comment by DeepakArora76** [comment]
@neil-119 Hi, I can see the problem with my sample app. I am looking into it. 

***** Fatal JavaScript exception - application has been terminated. *****
NativeScript encountered a fatal error: Uncaught Error: Building UI from XML. @app-root.xml:1:1
> NativeClass is not defined

**Comment by DeepakArora76** [comment]
@neil-119 Hi :) Thank you for reporting the issue. The problem was my end while building the plugin. I missed the setup that patches TS. There is a command that I must run before packing and publish the plugin. The command is "npm run setup", and I missed to fire that :)

I have released a new plugin:  3.5.0: Fixed runtime errors related to "NativeClass is not defined" observed on NativeScript Version 7 apps. The fix will likely benefit iOS apps.

I hope that this release fixes the problem.

Please test and post your feedback.
